sklearn/impute/tests/test_impute.py

```diff
@@ -971,42 +971,39 @@ def test_iterative_imputer_catch_warning():
+@pytest.mark.parametrize("min_value, max_value, correct_output",
+                         [(0,
+                           100,
+                           np.array([[0] * 3, [100] * 3])),
+                          (None,
+                           None,
+                           np.array([[-np.inf] * 3, [np.inf] * 3])),
+                          (-np.inf,
+                           np.inf,
+                           np.array([[-np.inf] * 3, [np.inf] * 3])),
+                          ([-5, 5, 10],
+                           [100, 200, 300],
+                           np.array([[-5, 5, 10], [100, 200, 300]])),
+                          ([-5, None, 10],
+                           [100, 200, None],
+                           np.array([[-5, -np.inf, 10], [100, 200, np.inf]]))])
+def test_iterative_imputer_min_max_completion(min_value,
+                                              max_value,
+                                              correct_output):
     # Test for issue #16349 which adds the capability
     # to specify different min and max values for all features.
     X = np.random.random((10, 3))
-    n_features = X.shape[1]
     imputer = IterativeImputer(min_value=min_value, max_value=max_value)
     imputer.fit(X)
     # Check if min_value and max_value have been cast to np arrays of
     # shape = n_features
     assert (isinstance(imputer._min_value, np.ndarray) and
-            (imputer._min_value.shape[0] == n_features))
+            (imputer._min_value.shape[0] == X.shape[1]))
     assert (isinstance(imputer._max_value, np.ndarray) and
-            (imputer._max_value.shape[0] == n_features))
-    # Make a dict with the correct values that each pair
-    # of min and max values should be cast to
-    correct_vals_dict = (
-        {(0, 100): np.array([[0] * n_features,
-                             [100] * n_features]),
-         (None, None): np.array([[-np.inf] * n_features,
-                                 [np.inf] * n_features]),
-         (-np.inf, np.inf): np.array([[-np.inf] * n_features,
-                                      [np.inf] * n_features]),
-         ((-5, 5, 10), (100, 200, 300)): np.array([[-5, 5, 10],
-                                                   [100, 200, 300]]),
-         ((-5, None, 10), (100, 200, None)): np.array([[-5, -np.inf, 10],
-                                                       [100, 200, np.inf]])}
-    )
-    # And check if the imputer has correctly processed the min and max values
-    assert_array_equal(correct_vals_dict[(min_value, max_value)][0, :],
-                       imputer._min_value)
-    assert_array_equal(correct_vals_dict[(min_value, max_value)][1, :],
-                       imputer._max_value)
+            (imputer._max_value.shape[0] == X.shape[1]))
+    # Check if the imputer has correctly processed the min and max values
+    assert_array_equal(correct_output[0, :], imputer._min_value)
```

Comment:
if we are going to have `inf`, the dtype of the array for comparison will be float and therefore we should use `assert_allclose` instead of `assert_array_equal`.

Reply:
Done!